documentation/src/test/java/org/hibernate/validator/referenceguide/chapter06/constraintvalidatorpayload/ConstraintValidatorPayloadTest.java

```diff
@@ -0,0 +1,48 @@
+package org.hibernate.validator.referenceguide.chapter06.constraintvalidatorpayload;
+
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+
+import org.hibernate.validator.HibernateValidator;
+import org.hibernate.validator.HibernateValidatorFactory;
+import org.junit.Test;
+
+@SuppressWarnings("unused")
+public class ConstraintValidatorPayloadTest {
+
+	@Test
+	public void setConstraintValidatorPayloadDuringValidatorFactoryInitialization() {
+		//tag::setConstraintValidatorPayloadDuringValidatorFactoryInitialization[]
+		ValidatorFactory validatorFactory = Validation.byProvider( HibernateValidator.class )
+				.configure()
+				.constraintValidatorPayload( "US" )
+				.buildValidatorFactory();
+
+		Validator validator = validatorFactory.getValidator();
+		//end::setConstraintValidatorPayloadDuringValidatorFactoryInitialization[]
+	}
+
+	@Test
+	public void setConstraintValidatorPayloadInValidatorContext() {
+		//tag::setConstraintValidatorPayloadInValidatorContext[]
+		HibernateValidatorFactory hibernateValidatorFactory = Validation.byDefaultProvider()
+				.configure()
+				.buildValidatorFactory()
+				.unwrap( HibernateValidatorFactory.class );
+
+		Validator validator = hibernateValidatorFactory.usingContext()
+				.constraintValidatorPayload( "US" )
```

Comment:
Now that I'm seeing the API used, I'm wondering whether it should be `constraintValidatorPayload( String.class, "US" )`. I can see two advantages:

* It makes it clearer that there can be multiple payloads (for different types)
* One can register a given implementation type under the key of a (public) interface

Btw. what happens if I register multiple payloads of the same type? Last one wins, or is an exception raised?

Reply:
Hum, no, it seems the implementation actually doesn't allow what I had in mind. Wondering whether that's not a bit too limited?

Reply:
We had the exact same discussion about the other payloads recently and you answered me the user could use a Map or a bean of some sort.

I don't think it's limited and it's consistent with the other payloads in the API.

Reply:
> ...and you answered me the user could use a Map or a bean of some sort.
>
> I don't think it's limited and it's consistent with the other payloads in the API.

I fully agree.